File: nodejs/activity.js
class Node {
    constructor(seatNo, personName) {
        this.seatNo = seatNo;
        this.personName = personName;
        this.next = null;
    }
}

class SeatingTicket {
    constructor() {
        this.head = null;
    }

    addSeat(seatNo, personName) {
        const newNode = new Node(seatNo, personName);
        
        // If the list is empty or the new node should be the new head
        if (this.head === null || this.head.seatNo >= seatNo) {
            newNode.next = this.head;
            this.head = newNode;
        } else {
            // Traverse the list to find the correct insertion point
            let currNode = this.head;
            while (currNode.next !== null && currNode.next.seatNo < seatNo) {
                currNode = currNode.next;
            }
            newNode.next = currNode.next;
            currNode.next = newNode;
        }
    }

    traverse() {
        let currNode = this.head;
        while (currNode !== null) {
            console.log(currNode.seatNo, currNode.personName);
            currNode = currNode.next;
        }
    }
}

// Example usage
const seating = new SeatingTicket();
seating.addSeat(10, 'Rishab');
seating.addSeat(20, 'Pandit');
seating.addSeat(15, 'naman');
seating.traverse();

File: python/dsa/activity.py
class Node:
    def __init__(self, seatNo, personName):
        self.seatNo = seatNo
        self.next = None
        self.personName = personName
 
class SeatingTicket:
    def __init__(self):
        self.head  = None
 
    def add_seat(self,seatNo, personName):
        new_node = Node(seatNo, personName)
        if self.head is None  or self.head.seatNo >= seatNo:
            new_node.next = self.head
            self.head = new_node
        else:
            curr_node = self.head
            while(curr_node.next is not None and curr_node.next.seatNo < new_node.seatNo):
                curr_node = curr_node.next
            new_node.next = curr_node.next
            curr_node.next = new_node
 
    def traverse(self):
        curr_node = self.head
        while(curr_node):
            print(curr_node.seatNo, curr_node.personName)
            curr_node = curr_node.next
               
seating = SeatingTicket()
seating.add_seat(10, 'Rishab')
seating.add_seat(20, 'Pandit')
seating.add_seat(15, 'naman')
seating.traverse()